main/explicit_bzero.c

```diff
@@ -30,15 +30,28 @@
 
 #include <string.h>
 
+#ifdef HAVE_ATTRIBUTE_WEAK
 __attribute__((weak)) void
 __explicit_bzero_hook(void *dst, size_t siz)
 {
 }
+#endif
 
 PHPAPI void php_explicit_bzero(void *dst, size_t siz)
 {
+#ifdef HAVE_ATTRIBUTE_WEAK
 	memset(dst, 0, siz);
 	__explicit_bzero_hook(dst, siz);
+#ifdef __GNUC__
+	asm __volatile__("" :: "r"(dst) : "memory"); 
+#endif
```

Comment:
After looking around a bit, I assume the problem being fixed here is that the weak symbol reference may be optimized away in LTO builds, and having the asm barrier prevents that. I wonder if we should just drop the `explicit_bzero_hook` entirely and only use the asm barrier? It seems that it only works for OpenBSD because they know that LTO is not used.

Reply:
I thought of that too just tried to do not be to far away from upstream but if not s an issue then ...